### PR TITLE
additional logging catch statements added to fix #4315

### LINKED ISSUE #4315: Restore fails when creating empty files at the end of restore process
- [ x] I have searched open and closed issues for duplicates.

----------------------------------------

## Environment info

- **Duplicati version**: 2.0.5.110_canary_2020-08-10
- **Operating system**: Windows 10
- **Backend**: local usb3 drive

## Description
When trying to restore my backup taken under osx the restore seems to be able to restore files but then fails at the end of the restore process with

> System.IO.IOException: Die Syntax für den Dateinamen, Verzeichnisnamen oder die Datenträgerbezeichnung ist falsch.
>    bei System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
>    bei System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
>    bei System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize)
>    bei Duplicati.Library.Common.IO.SystemIOWindows.FileCreate(String path)
>    bei Duplicati.Library.Main.Operation.RestoreHandler.DoRun(LocalDatabase dbparent, IFilter filter, RestoreResults result)
>    bei Duplicati.Library.Main.Operation.RestoreHandler.Run(String[] paths, IFilter filter)
>    bei Duplicati.Library.Main.Controller.<>c__DisplayClass15_0.<Restore>b__0(RestoreResults result)
>    bei Duplicati.Library.Main.Controller.RunAction[T](T result, String[]& paths, IFilter& filter, Action`1 method)
>    bei Duplicati.Library.Main.Controller.Restore(String[] paths, IFilter filter)
>    bei Duplicati.Server.Runner.Run(IRunnerData data, Boolean fromQueue)

Would it be possible to add some debug logging so that it is possible to identify the files that are causing the issue?

### PR DESCRIPTION
**Pull request by barfoo4711**
fixed #4315 by adding catching of exceptions when creating empty files Not sure if it is ok to create WARN messages or if it should be ERROR messages instead

### CODE REVIEW

File: Duplicati/Library/Main/Operation/RestoreHandler.cs [hunk 1]
```diff
@@ -415,11 +415,21 @@ private void DoRun(LocalDatabase dbparent, Library.Utility.IFilter filter, Resto
...
+                    {
+                        SystemIO.IO_OS.DirectoryCreate(SystemIO.IO_OS.PathGetDirectoryName(file.Path));
+                        // Just create the file and close it right away, empty statement is intentional.
+                        using (SystemIO.IO_OS.FileCreate(file.Path))
+                        {
+                        }
+                    }
+                    catch (Exception ex)
                     {
+                        Logging.Log.WriteWarningMessage(LOGTAG, "RestoreFileFailed", ex, "Failed to restore empty file: \"{0}\", message: {1}", file.Path, ex.Message);
```
Comment (warwickmm) [inline Duplicati/Library/Main/Operation/RestoreHandler.cs:430]: For non-empty files, we write an error message, and also increment the `fileErrors` variable.  Should we do the same here?

https://github.com/duplicati/duplicati/blob/9b977b7ba04a940868754fd15a37b99fbd77f1bd/Duplicati/Library/Main/Operation/RestoreHandler.cs#L472-L478

File: Duplicati/Library/Main/Operation/RestoreHandler.cs [hunk 2]
```diff
@@ -415,11 +415,21 @@ private void DoRun(LocalDatabase dbparent, Library.Utility.IFilter filter, Resto
                 // Restore empty files. They might not have any blocks so don't appear in any volume.
                 foreach (var file in database.GetFilesToRestore(true).Where(item => item.Length == 0))
                 {
-                    SystemIO.IO_OS.DirectoryCreate(SystemIO.IO_OS.PathGetDirectoryName(file.Path));
+                    Logging.Log.WriteProfilingMessage(LOGTAG, "RestoreFile", "restoring empty file \"{0}\"", file.Path);
```
Comment (warwickmm) [inline Duplicati/Library/Main/Operation/RestoreHandler.cs:418]: Can we capitalize the first `r` in `restoring`?

**Comment by warwickmm** [comment]
Thanks @barfoo4711 for the pull request.  I've gone ahead and made a few additional changes to your branch.  I think we can merge now.

**Comment by barfoo4711** [comment]
Thank you very much for fixing and sorry for not reacting on your request :( I'm a Java guy so I'm not really familiar with the .net related stuff :( But will retest with my osx backup once this fix is released in the canary channel!

**Pull request merged**